systems-state-manager/nisystemsstate.yml

```diff
@@ -0,0 +1,999 @@
+        type: string
+        description: Name of the package.
+      version:
+        type: string
+        description: Version of the package.
+    required:
+      - name
+    example:
+      name: vim
+      version: '7.1'
+
+paths:
+  /:
+    get:
+      operationId: HandleGetVersionInfo
+      x-ni-request-all-privileges: true
+      summary: API information
+      tags: [versioning]
+      security: []
+      description: Gets permissions and operations for the StatesService.
+      responses:
+        200:
+          description: OK
+          schema:
+            description: Version information.
+            title: RootEndpointResponse
+            type: object
+            properties:
+              v1:
+                $ref: '#/definitions/V1Operations'
+              version:
+                description: The implementation version of the web service.
+                type: string
+        default:
+            $ref: '#/responses/Error'
+  /{version}:
+    get:
+      operationId: RootEndpointWithVersion
+      x-ni-request-all-privileges: true
+      summary: API version information
+      description: Returns available operations for a single version of the API.
+      tags: [versioning]
+      security: []
+      parameters:
+        - in: path
+          name: version
+          description: The version of the API to retrieve operations.
+          x-example: v1
+          type: string
+          required: true
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: '#/definitions/V1Operations'
+        404:
+          description: Not Found
+        default:
+            $ref: '#/responses/Error'
+  /v1/states:
+    get:
+      operationId: GetStates
+      x-ni-privilege: Read
+      x-ni-operation: getStates
+      tags: [states]
+      summary: Get all system states
+      description: Return metadata information of all system states.
+      parameters:
+        - in: query
+          name: skip
+          description: How many states to skip. Used for pagination.
+          type: integer
+          x-ni-data-type: U32
+          default: 0
+        - in: query
+          name: take
+          description: How many states to retrieve. Used for pagination.
+          type: integer
+          x-ni-data-type: U32
+          default: 1000
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: '#/definitions/StateDescriptionListResponse'
+        401:
+          $ref: '#/responses/Unauthorized'            
+        default:
+          $ref: '#/responses/Error'
+    post:
+      operationId: CreateState
+      x-ni-privilege: nisystemsstate.Write
+      x-ni-operation: createOrUpdateStates
+      tags: [states]
+      summary: Create a new system state
+      description: Create a new System State by providing metadata and its content. rawState property will be used if it has a non-null value, feeds and packages will be used otherwise.
```

Comment:
"Create a new system state by providing the required metadata and content. If you provide a non-null value for the rawState property, it will be used. Otherwise, feeds and packages will be used."?